tests/wasi.rs

```diff
@@ -120,9 +120,27 @@ impl crate::wasi_snapshot_preview1::WasiSnapshotPreview1 for WasiCtx {
     fn fd_pread(
         &self,
         _fd: types::Fd,
-        _iovs: &types::IovecArray<'_>,
+        iovs: &types::IovecArray<'_>,
         _offset: types::Filesize,
     ) -> Result<types::Size> {
+        // This is not functional code, but the type annotations demonstrate
+        // that we can use the wiggle API to create the datastructures we want
+        // for efficient implementation of this function elsewhere.
+
+        let mut bc = GuestBorrows::new();
+        let mut slices: Vec<&'_ mut [u8]> = Vec::new();
+        for iov_ptr in iovs.iter() {
+            let iov: types::Iovec = iov_ptr
+                .expect("iovec element pointer is valid")
+                .read()
+                .expect("read iovec element");
+            let base: GuestPtr<u8> = iov.buf;
+            let len: u32 = iov.buf_len;
+            let buf: GuestPtr<[u8]> = base.as_array(len);
+            let slice = buf.as_raw(&mut bc).expect("borrow slice from iovec");
+            slices.push(unsafe { &mut *slice });
```

Comment:
Scrutinizing this a bit the `GuestBorrows` types doesn't include the `types::IovecArray` itself, so this could in theory be aliasing I believe? Definitely a tricky case :(

Reply:
you're correct, it does not, so it could indeed be aliasing. That is one downside of doing the borrow checking manually, versus centralized like before. I'm not sure what, if anything, we can do about this.